spec/classes/init_spec.rb

```diff
@@ -33,17 +33,32 @@
 
   osfamily_matrix = {
     'Debian-7' => {
-      :architecture           => 'x86_64',
-      :osfamily               => 'Debian',
-      :operatingsystemrelease => '7',
-      :ssh_version            => 'OpenSSH_6.0p1',
-      :ssh_version_numeric    => '6.0',
-      :ssh_packages           => ['openssh-server', 'openssh-client'],
-      :sshd_config_mode       => '0600',
-      :sshd_service_name      => 'ssh',
-      :sshd_service_hasstatus => true,
-      :sshd_config_fixture    => 'sshd_config_debian',
-      :ssh_config_fixture     => 'ssh_config_debian',
+      :architecture              => 'x86_64',
+      :osfamily                  => 'Debian',
+      :operatingsystemrelease    => '7',
+      :os.release.major          => '7',
+      :ssh_version               => 'OpenSSH_6.0p1',
+      :ssh_version_numeric       => '6.0',
+      :ssh_packages              => ['openssh-server', 'openssh-client'],
+      :sshd_config_mode          => '0600',
+      :sshd_service_name         => 'ssh',
+      :sshd_service_hasstatus    => true,
+      :sshd_config_fixture       => 'sshd_config_debian',
+      :ssh_config_fixture        => 'ssh_config_debian',
+    },
+    'Debian-9' => {
+      :architecture              => 'x86_64',
+      :osfamily                  => 'Debian',
+      :operatingsystemrelease    => '9',
+      :os.release.major          => '9',
```

Comment:
I think this doesn't result in the key you want, I don’t think symbols can be dot expanded like that. You probably need the right hand side to be a hash with release as the key and a hash value with major as the key.
Alternatively, switch to rspec-puppet-facts, but that may be better served as its own separate PR.

Reply:
```ruby
:os => {
  :release => {
    major => '9',
  },
},
```

Reply:
thanks guys. But it fails with this syntax.